src/v/container/printable_range.h

```diff
@@ -0,0 +1,33 @@
+/*
+ * Copyright 2024 Redpanda Data, Inc.
+ *
+ * Use of this software is governed by the Business Source License
+ * included in the file licenses/BSL.md
+ *
+ * As of the Change Date specified in that file, in accordance with
+ * the Business Source License, use of this software will be governed
+ * by the Apache License, Version 2.0
+ */
+#pragma once
+#include <iostream>
+#include <utility>
+
+template<class Range>
+class printable_range {
```

Comment:
this class seems to be redundant as we already have ` fmt::join` which does this.